### PR TITLE
separate RSpec handling in runner

### PR DESCRIPTION
**Pull request by arlimus**
This has so far prevented us from writing some critical unit tests for profile structure and test loading. By separating it and adding a Mock runner we can now test for profile structure. (more MRs to come)

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by srenatus** [comment]
_sigh_ turning down concurrency solves the issue locally, but triggers a timeout (no output for 10 minutes) on Travis. :unamused:

**Comment by srenatus** [comment]
It's not a train issue after all, https://travis-ci.org/chef/inspec/jobs/102517662 uses 0.9.4, too, and I've just re-run its tests successfully.

File: lib/inspec/resource.rb [hunk 1]
```diff
@@ -11,8 +11,31 @@ class Resource
...
       @registry ||= {}
     end
+
+    # Creates the inner DSL which includes all resources for
+    # creating tests. It is always connected to one target,
+    # which is specified via the backend argument.
+    #
+    # @param backend [BackendRunner] exposing the target to resources
+    # @return [ResourcesDSL]
+    def self.create_dsl(backend)
```
Comment (chris-rock) [inline lib/inspec/resource.rb:21]: much better to have the dsl in here ...

**Comment by chris-rock** [comment]
@arlimus awesome work to bring rspec under control and to enable better unit testing. :100: 
@srenatus I reversed your commit, travis is green again...

**Comment by srenatus** [comment]
@chris-rock thanks a lot! Any ideas what had happened here?

**Pull request merged**